### ISSUE TITLE
Question: is there a way to disclose our private channels to certain peers?

### ISSUE THREAD

**Issue by Overtorment** [lightningnetwork/lnd#5085]
Use case 1: I want an external service to plot a route for me accounting for my private channels
Use case 2: I want certain nodes to be able to route through my private channels, but keep those channels private for general network

**Comment by Roasbeef** [comment]
In theory, you should be able to make a custom invoice that has hop hints which may not be directly adjacent to your node. This would let you create a sort of backbone private network, which is what I think you're after here. The `AddInvoice` call lets you set a set of custom route hints: https://api.lightning.community/#addinvoice

If the invoice is created w/ the `invoice.Private` flag, then it'll _combine_ the hop hints that were passed in with its own hop hints: https://github.com/lightningnetwork/lnd/blob/4bab68a8085cd38526e62bc701725443631db5da/lnrpc/invoicesrpc/addinvoice.go#L278

**Comment by Roasbeef** [comment]
> What I really want is an external node that has no channels with me to be able to plot a route for me and my private channels via QueryRoutes

In theory this should be possible if you use `QueryRoutes`, but set the `SourcePubKey` to the node with the private channels, and the destination as normal. It should then treat the provided hop hints just as extra edges in the graph and work as expected*. 

> And second use case is that I want certain node that has no channels with me to be able to route through my private channels

This second case should also be possible if you, give them a route that already includes those private channels. This stuff _should_ already work, but we haven't explicitly tested these scenarios. 

What've you tried so far?

**Comment by Overtorment** [comment]
> In theory this should be possible if you use QueryRoutes, but set the SourcePubKey to the node with the private channels, and the destination as normal. It should then treat the provided hop hints just as extra edges in the graph and work as expected*.

I've tried that, but that only works if client has _public_ channel. as soon as I set it to private - queryRoutes on 3rd party node starts to fail.
I've built an API for this: http://lndhub.herokuapp.com/queryroutes/03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f/03c2abfa93eacec04721c019644584424aab2ba4dff3ac9bdab4e9c97007491dda/100
this API can plot a route from A to B, but only through public channels. if A has only private channels - this will fail.


example topology:

mobile client has private channel with ACINQ.
ACINQ has public channel with TIPPINME.
LNDHUB may or may not have channels with ACINQ or TIPPINME, but actively syncs graph.

mobile client asks LNDHUB to plot a route from himself to TIPPINME.
actual result: fail, because client's channel is private.

I know a workaround for this, that is, ask LNDHUB to plot a route from ACINQ to TIPPINME, and then manually add a hop from myself to ACINQ, but
that feels hacky
